### PR TITLE
Use libblkid to check swap status before swapon

### PR DESCRIPTION
**Pull request by vojtechtrefny**
libblkid probe is more reliable than our custom check.

### CODE REVIEW

File: src/plugins/swap.c [hunk 1]
```diff
@@ -198,56 +201,150 @@ gboolean bd_swap_swapon (const gchar *device, gint priority, GError **error) {
...
         bd_utils_report_finished (progress_id, (*error)->message);
+        blkid_free_probe (probe);
         return FALSE;
     }
 
-    page_size = getpagesize ();
-    page_size = MAX (2048, page_size);
-    io_status = g_io_channel_seek_position (dev_file, page_size - 10, G_SEEK_SET, &tmp_error);
-    if (io_status != G_IO_STATUS_NORMAL) {
+    /* we may need to try mutliple times with some delays in case the SWAPDEVICE is
```
Comment (vpodzime) [inline src/plugins/swap.c]: `the device`, no `SWAPDEVICE` in this codebase ;)

File: src/plugins/swap.c [hunk 2]
```diff
@@ -198,56 +201,150 @@ gboolean bd_swap_swapon (const gchar *device, gint priority, GError **error) {
...
         return FALSE;
     }
 
-    io_status = g_io_channel_read_chars (dev_file, dev_status, 10, &num_read, &tmp_error);
-    if ((io_status != G_IO_STATUS_NORMAL) || (num_read != 10)) {
+    blkid_probe_enable_partitions (probe, 1);
+    blkid_probe_set_partitions_flags (probe, BLKID_PARTS_MAGIC);
+    blkid_probe_enable_superblocks (probe, 1);
+    blkid_probe_set_superblocks_flags (probe, BLKID_SUBLKS_USAGE | BLKID_SUBLKS_TYPE |
+                                              BLKID_SUBLKS_MAGIC | BLKID_SUBLKS_BADCSUM);
```
Comment (vpodzime) [inline src/plugins/swap.c]: Do we really want these features? Not sure what *superblocks* here are, but we are definitely not probing for partitions.

File: src/plugins/swap.c [hunk 3]
```diff
@@ -198,56 +201,150 @@ gboolean bd_swap_swapon (const gchar *device, gint priority, GError **error) {
...
+        close (fd);
         return FALSE;
     }
 
-    g_io_channel_shutdown (dev_file, FALSE, &tmp_error);
-    g_clear_error (&tmp_error);
+    status = blkid_probe_lookup_value (probe, "TYPE", &value, NULL);
+    if (status != 0) {
+        g_set_error (error, BD_SWAP_ERROR, BD_SWAP_ERROR_UNKNOWN_STATE,
+                     "Failed to get filesystem type for the device '%s'", device);
```
Comment (vpodzime) [inline src/plugins/swap.c]: *format type*

File: src/plugins/swap.c [hunk 4]
```diff
@@ -198,56 +201,150 @@ gboolean bd_swap_swapon (const gchar *device, gint priority, GError **error) {
...
+        bd_utils_report_finished (progress_id, (*error)->message);
+        blkid_free_probe (probe);
+        close (fd);
+        return FALSE;
+    }
 
-    if (g_str_has_prefix (dev_status, "SWAP-SPACE")) {
+    if (g_strcmp0 (value, "swap") != 0) {
+        g_set_error (error, BD_SWAP_ERROR, BD_SWAP_ERROR_UNKNOWN_STATE,
+                     "Device '%s' is not a swap", device);
```
Comment (vpodzime) [inline src/plugins/swap.c]: *is not formatted as swap*

File: src/plugins/swap.c [hunk 5]
```diff
@@ -198,56 +201,150 @@ gboolean bd_swap_swapon (const gchar *device, gint priority, GError **error) {
...
                      "Unknown swap space format, cannot activate.");
         bd_utils_report_finished (progress_id, (*error)->message);
+        blkid_free_probe (probe);
+        close (fd);
         return FALSE;
     }
 
+    status_len = (gint64) strlen (value);
+
+    status = blkid_probe_lookup_value (probe, "SBMAGIC_OFFSET", &value, NULL);
```
Comment (vpodzime) [inline src/plugins/swap.c]: I think this is the last place the `probe` is needed so it should be free'd here.

Comment (vojtechtrefny) [reply src/plugins/swap.c]: `value` is owned by the probe, so we can't free it here.

**Pull request merged**